### PR TITLE
[Dashboard] Increase the RPC timeout for the snapshot API

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

The current RPC timeout is too short (2s), and we've discovered Ray components not responding within the current timeout range occasionally under pressure. This is something we will fix, but it's better if we can have a longer timeout. The current timeout is configured as 2X of the polling frequency. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by wuisawesome — DISMISSED** [review]
Can we make this configurable via env var?

**Comment by wuisawesome** [comment]
On a somewhat related note, I wonder if we should distinguish between timeouts and other error types...

File: dashboard/modules/snapshot/snapshot_head.py
Comment (galenhwang) [inline dashboard/modules/snapshot/snapshot_head.py:31]: Do we want to pass timeout in as a query param for any of these? Similar to the `component_activities` route.

https://github.com/ray-project/ray/blob/de2db06d182aa68a368007db4582087435e3c09a/dashboard/modules/snapshot/snapshot_head.py#L132-L136

That way, if any clusters have long snapshots, we could use a feature flag to target specific clusters to extend the timeout period.

**Review by galenhwang — APPROVED** [review]

**Comment by jjyao** [comment]
I may miss some context here: will snapshot API be replaced completely by `component_activities`. Anything we do for snapshot, should we consider it for `component_activities` as well?

**Comment by wuisawesome** [comment]
Actually, also are we sure we need this in light of https://github.com/ray-project/ray/pull/27589?

**Comment by rkooo567** [comment]
@wuisawesome it is still safe to do this. I think 2 seconds are too drastic timeout in general

**Comment by rkooo567** [comment]
@jjyao let me double check if this change is reflected to the component activities too.

**Comment by rkooo567** [comment]
@wuisawesome @galenhwang I will go with Galen's suggestion for the timeout configuration (it can be specified via Http req argument)

**Comment by galenhwang** [comment]
Would we able to backport this to previous Ray versions? This mainly affects pre-2.0 versions.

**Comment by rkooo567** [comment]
@matthewdeng when do we plan to do the minor release for 2.0,1?

**Comment by rkooo567** [comment]
cc @matthewdeng to answer https://github.com/ray-project/ray/pull/28330#issuecomment-1245757981

**Comment by rkooo567** [comment]
@jjyao confirms it also affects timeout for `component_activities `. I think snapshot is still used along with `component_activities `

**Comment by matthewdeng** [comment]
We could perform a patch release for older versions (perhaps just 1.13), but would be hesitant to do so unless there is a pressing need.

**Comment by rkooo567** [comment]
Added the timeout support. I will try merging it once the CI passes.

**Comment by rkooo567** [comment]
cc @alanwguo I need the code owner approval too

**Review by alanwguo — APPROVED** [review]

**Pull request merged**
